### ISSUE TITLE
autofocus: on autocomplete inside a dialog bug

### ISSUE THREAD

**Issue by EladBezalel** [angular/material#5776]
In [this Codepen](http://codepen.io/EladBezalel/pen/MaLGJe)

When clicking on the button the dialog is opened with the focused autocomplete but the list of options is opened from the button.

**Comment by DevVersion** [comment]
Hello,

I tested a bit with the dialogs and the autocomplete.
The thing is we should focus the Elements after the dialog is open, not at postLink as you can see here
```
    $mdUtil.nextTick(function () {
      gatherElements();
      moveDropdown();
      focusElement();
      $element.on('focus', focusElement);```

I already tried to do this in the dialog.js but thats only working over the directive controller.

So you have any idea in fixing that? I'm available in creating a PR

**Comment by EladBezalel** [comment]
@DevVersion not really got into this, as always a PR is very welcomed

**Comment by DevVersion** [comment]
I'm still trying to get into that... we should init the object after the dialog got opened, how we can throttle that? Any ideas?

**Comment by EladBezalel** [comment]
Solved in #5943 kudos to @DevVersion

**Comment by DevVersion** [comment]
@EladBezalel This isn't happening anymore? Closing?
